app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,43 @@
 package com.lightricks.feedexercise.data
 
+import androidx.lifecycle.Transformations
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedDbEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.FeedData
+import io.reactivex.Completable
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
+class FeedRepository(private val database: FeedDatabase, private val api: FeedApiService) {
+    fun refresh(): Completable {
```

Comment:
This is a public method.
If somebody will call `refresh` from the main thread it will crash (writing to DB on main thread).
You should either add `.subscribeOn(Schedulers.io())` here or you should mention it in the documentation.

Reply:
Nope. That's wrong. RxJava scheduler returns a promise, it does not perform it synchronously. It runs on Scheduler.io() context (as can be seen when putting a breakpoint).
Renamed it to refreshAsync() - to match the Coroutines conventions of returning async objects.

Reply:
My comment was not clear. 
My point is that the caller doesn't know what `FeedRepository` is going to do behind the scene and he doesn't know that scheduling on the main thread is prohibited. So it's `FeedRepository`'s responsibility to either subscribe on a background thread OR to document that the caller has to do it.